quarkusio/quarkusio.github.io
# Search through https://search.quarkus.io

Creating as draft, for discussion. Please do not merge yet.

Summary of the changes:

* Search in `/guides` and `/version/*/guides` is now implemented with a Vue.js application, replacing the custom Javascript of `search-filter.js`.
* Search is performed remotely by https://github.com/quarkusio/search.quarkus.io
* When https://github.com/quarkusio/search.quarkus.io fails to respond in a reasonable time (currently 400ms) or errors out, we default to Javascript search (basically the same implementation than before, just without JQuery)
* When searching, we no longer display anything else than matching cards (no category headings in particular). This allows ordering hits by score (relevance to the searched terms).

To test locally:

* Start https://github.com/quarkusio/search.quarkus.io locally (`quarkus dev`)
* Run the website (`./serve.sh` or `docker compose up`)
* Do a search at http://localhost/guides

You should notice search results are slightly more relevant, though we didn't try to tune search very much for now. You will see improvements though, e.g. matches against content that is not in the title/summary but elsewhere in the guide
